python-sdk/tests/test_example_dags.py

```diff
@@ -51,44 +51,44 @@ def session():
     return get_session()
 
 
-DAG_BAG = DagBag(Path(__file__).parent.parent / "example_dags", include_examples=False)
-AIRFLOW_VERSION_INDICATOR = "airflow_version:"
-MINIMUM_AIRFLOW_VERSION = "2.2.5"
+MIN_VER_DAG_FILE: dict[str, list[str]] = {
+    "2.3": ["example_dynamic_task_template.py", "example_bigquery_dynamic_map_task.py"],
+    "2.4": ["example_datasets.py"],
+}
 
+# Sort descending based on Versions and convert string to an actual version
+MIN_VER_DAG_FILE_VER: dict[Version, list[str]] = {
+    Version(version): MIN_VER_DAG_FILE[version]
+    for version in sorted(MIN_VER_DAG_FILE, key=Version, reverse=True)
+}
 
-def get_airflow_version(dag: DAG) -> str:
-    for tag in dag.tags:
-        if tag.startswith(AIRFLOW_VERSION_INDICATOR):
-            return tag[len(AIRFLOW_VERSION_INDICATOR) :]
-    return MINIMUM_AIRFLOW_VERSION
 
+def get_dag_bag() -> DagBag:
+    """Create a DagBag by adding the files that are not supported to .airflowignore"""
+    example_dags_dir = Path(__file__).parent.parent / "example_dags"
+    airflow_ignore_file = example_dags_dir / ".airflowignore"
 
-def get_airflow_dags() -> Iterator[tuple[str, str]]:
-    for dag_id, dag in DAG_BAG.dags.items():
-        yield dag_id, dag, get_airflow_version(dag)
+    with open(airflow_ignore_file, "w+") as file:
+        for min_version, files in MIN_VER_DAG_FILE_VER.items():
+            if Version(airflow.__version__) < min_version:
+                print(f"Adding {files} to .airflowignore")
+                file.writelines([f"{file}\n" for file in files])
```

Comment:
This line looks a bit strange because you "redefine" `file`. 

Btw you could also just do this, or?
```diff
- file.writelines([f"{file}\n" for file in files])
+ file.write("\n".join(files)
```